examples/ppo/unit_tests.py

```diff
@@ -0,0 +1,87 @@
+
+import jax
+import flax
+from flax import nn
+import numpy as onp
+
+import numpy.testing as onp_testing
+from absl.testing import absltest
+
+#test GAE
+from main import gae_advantages
+class TestGAE(absltest.TestCase):
+  def test_gae_random(self):
+    # create random data, simulating 4 parallel envs and 20 time_steps
+    envs, steps = 10, 100
+    rewards = onp.random.choice([-1., 0., 1.], size=(steps, envs),
+                                p=[0.01, 0.98, 0.01])
+    terminal_masks = onp.ones(shape=(steps, envs), dtype=onp.float64)
+    values = onp.random.random(size=(steps + 1, envs))
+    discount = 0.99
+    gae_param = 0.95
+    adv = gae_advantages(rewards, terminal_masks, values, discount, gae_param)
+    self.assertEqual(adv.shape, (steps, envs))
+    # test the property A_{t} = \delta_t + \gamma*\lambda*A_{t+1}
+    # for each agent separately
+    for e in range(envs):
+      for t in range(steps-1):
+        delta = rewards[t, e] + discount * values[t+1, e] - values[t, e]
+        lhs = adv[t, e]
+        rhs = delta + discount * gae_param * adv[t+1, e]
+        onp_testing.assert_almost_equal(lhs, rhs)
+
+#test environment and preprocessing
+from remote import RemoteSimulator, rcv_action_send_exp
```

Comment:
all imports at the top of the file

Reply:
Resolved by https://github.com/google/flax/pull/470/commits/f53c1df003608da37e683891d1e1b5503fe3724a